src/Irc.hs

```diff
@@ -39,25 +39,32 @@ import           Network.IRC.CTCP             (CTCPByteString, asCTCP)
 import           Prelude                      hiding (and)
 import           System.Console.Concurrent    (outputConcurrent)
 import           System.IO.Error              (ioeGetErrorString)
+import           Control.Exception.Lifted
 
 import           Network.SimpleIRC
 
-connectTo :: IrcParams -> Bool -> IO () -> IO () -> IrcIO Connection
-connectTo params withDebug onConnected onJoined = do
+connectTo :: Options -> IrcParams -> Bool -> IO () -> IO () -> IrcIO Connection
+connectTo opt@Options{..} params withDebug onConnected onJoined = do
     (conf, bcs) <- lift $ config params
     con <- connect' conf withDebug
     lift $ mapM_ ((\f -> f con) . onConnect) (hooks params)
     lift onConnected
-    joined <- lift $ waitForAll bcs
+    joined <- lift $ catch (waitForAll bcs) (handler con)
```

Comment:
> bracket only guarantees that once a resource has been successfully acquired, any exception thrown in the 'apply' function will be caught and handled.

D'oh, of course! Thanks for the reminder. ;)

> i pressed Ctrl-C causing an async exception

I didn't bother handling this so far as most IRC networks didn't care about dropped connections, but certainly this should be handled. At the moment I'm trying to find a more concise way to do the exception handling.